packages/web5/lib/src/dids/data_models/did_document.dart

```diff
@@ -124,6 +127,34 @@ class DidDocument {
     service!.add(svc);
   }
 
+  DidResource? getResourceById(String resourceId) {
+    if (resourceId == id) {
+      return this;
+    }
+
+    final Set<String> idVariations = {resourceId};
+    if (resourceId.startsWith('#')) {
+      idVariations.add('$id$resourceId');
+    } else {
+      final splitId = resourceId.split('#');
+      if (splitId.length > 1) {
+        idVariations.add('#${splitId[1]}');
+      }
+    }
+
+    DidResource? resource =
+        verificationMethod?.firstWhere((vm) => idVariations.contains(vm.id));
+
+    if (resource != null) {
+      return resource;
+    }
+
+    resource = service?.firstWhere((vm) => idVariations.contains(vm.id));
```

Comment:
there's a `firstOrNullWhere` as well if it's possible `vm.id` is not found. With this code, it will throw if it's not found. With `firstOrNullWhere` it will just return `null`.

Reply:
ah nice! good call. didn't realize the collections package had fancy extensions. 

done! https://github.com/TBD54566975/web5-dart/commit/0c9dda6ac3898be74a8605d2ab5e4689b202729d